src/webgpu/api/validation/encoding/beginComputePass.spec.ts

```diff
@@ -191,3 +191,36 @@ g.test('timestamp_query_set,device_mismatch')
 
     t.tryComputePass(!mismatched, descriptor);
   });
+
+g.test('begin_compute_pass_before_ending_previous_pass')
+  .desc(`Test that beginning a compute pass before ending the previous pass causes an error.`)
+  .fn(async t => {
+    const encoder = t.device.createCommandEncoder();
+
+    const attachmentTexture = t.device.createTexture({
+      format: 'rgba8unorm',
+      size: { width: 16, height: 16, depthOrArrayLayers: 1 },
+      usage: GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+
+    const renderPass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          view: attachmentTexture.createView(),
+          clearValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+          loadOp: 'clear',
+          storeOp: 'store',
+        },
+      ],
+    });
+
+    // Begin a compute pass before ending the previous render pass.
+    const computePass = encoder.beginComputePass();
+
+    renderPass.end();
+    computePass.end();
```

Comment:
Because end() can also fail validation due to incorrect state, it's a bit tricky to test this robustly.
Let's add some extra subcases, maybe something like this:
```ts
    .combine('endPasses', [[], [0], [1], [0, 1], [1, 0]])
```
```ts
    const passes = [renderPass, computePass];
    for (const index of t.params.endPasses) {
        passes[index].end();
    }
```

to make sure we cover every state error we could reasonably detect, e.g. the implementation incorrectly does one of the following:
- implicitly closes the first pass
- ignores the second pass without setting an error
- allows the two passes to run in parallel
- allows passes to nest

Reply:
I apply the subcases to the test. And, it seems to me that separate tests should handle the error states.